### PR TITLE
Add survival rate parameter

### PR DESCRIPTION
**Pull request by wenzeslaus**
The addition of survival rate requires an additional parameter plus date settings. This adds it as a input raster series and generalizes code for reading raster names for raster series from a file which is common between survival rate and temperatures. Tests are for 100% and 0% survival rate.

Additionally, this adds number of steps as verbose message and fixes seed print (extra char d in output).

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -409,6 +437,39 @@ int main(int argc, char *argv[])
...
+    opt.survival_rate_month->description =
+        _("The survival rate is applied at selected month and day");
+    opt.survival_rate_month->required = NO;
+    opt.survival_rate_month->guisection = _("Weather");
+
+    opt.survival_rate_day = G_define_option();
+    opt.survival_rate_day->type = TYPE_INTEGER;
+    opt.survival_rate_day->key = "survival_day";
+    opt.survival_rate_day->label =
+        _("Day of selected month when the pest or patogen dies with given rate");
```
Comment (petrasovaa) [inline main.cpp]:
```suggestion
        _("Day of selected month when the pest or pathogen dies with given rate");
```

File: main.cpp [hunk 2]
```diff
@@ -409,6 +437,39 @@ int main(int argc, char *argv[])
     opt.weather_coefficient_file->required = NO;
     opt.weather_coefficient_file->guisection = _("Weather");
 
+    // Survival rate
+
+    opt.survival_rate_month = G_define_option();
+    opt.survival_rate_month->type = TYPE_INTEGER;
+    opt.survival_rate_month->key = "survival_month";
+    opt.survival_rate_month->label =
+        _("Month when the pest or patogen dies due to low temperature");
```
Comment (petrasovaa) [inline main.cpp]:
```suggestion
        _("Month when the pest or pathogen dies due to low temperature");
```

Comment (petrasovaa) [inline main.cpp]: Inconsistent description with survival day. Not sure if it always has to be due to low temperature.

File: main.cpp [hunk 3]
```diff
@@ -409,6 +437,39 @@ int main(int argc, char *argv[])
...
+        _("Day of selected month when the pest or patogen dies with given rate");
+    opt.survival_rate_day->description =
+        _("The survival rate is applied at selected month and day");
+    opt.survival_rate_day->required = NO;
+    opt.survival_rate_day->guisection = _("Weather");
+
+    opt.survival_rate_file = G_define_standard_option(G_OPT_F_INPUT);
+    opt.survival_rate_file->key = "survival_rate";
+    opt.survival_rate_file->label =
+        _("Input file with one suvival rate raster map name per line");
```
Comment (petrasovaa) [inline main.cpp]:
```suggestion
        _("Input file with one survival rate raster map name per line");
```

File: main.cpp [hunk 4]
```diff
@@ -857,6 +919,13 @@ int main(int argc, char *argv[])
         config.mortality_frequency_n = opt.mortality_frequency_n->answer ? std::stoi(opt.mortality_frequency_n->answer) : 0;
     }
 
+    if (opt.lethal_temperature_months->answer)
+        config.survival_rate_month = std::stoi(opt.survival_rate_month->answer);
+    if (opt.lethal_temperature->answer)
+        config.survival_rate_day = std::stod(opt.survival_rate_day->answer);
+    if (opt.survival_rate_file->answer)
+        config.use_survival_rate = true;
+
```
Comment (petrasovaa) [inline main.cpp]:
```suggestion
    if (opt.survival_rate_month->answer)
        config.survival_rate_month = std::stoi(opt.survival_rate_month->answer);
    if (opt.survival_rate_day->answer)
        config.survival_rate_day = std::stod(opt.survival_rate_day->answer);
    if (opt.survival_rate_file->answer)
        config.use_survival_rate = true;

```

**Review by petrasovaa — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (wenzeslaus) [reply main.cpp]: Sorry, more a copy-paste issue. Made it general like the day description.

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**